internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -1528,11 +1531,72 @@ func (r *ClusterDeploymentsReconciler) SetupWithManager(mgr ctrl.Manager) error
 		}
 	}
 
+	mapAgentToClusterDeployment := func(a client.Object) []reconcile.Request {
+		log := logutil.FromContext(context.Background(), r.Log).WithFields(
+			logrus.Fields{
+				"agent":           a.GetName(),
+				"agent_namespace": a.GetNamespace(),
+			})
+		agent, ok := a.(*aiv1beta1.Agent)
+		if !ok {
+			log.Errorf("%v was not an Agent", a) // shouldn't be possible
+			return []reconcile.Request{}
+		}
+		if agent.Spec.ClusterDeploymentName != nil {
+			log.Debugf("Map Agent : %s %s CD ref name %s ns %s", agent.Namespace, agent.Name, agent.Spec.ClusterDeploymentName.Name, agent.Spec.ClusterDeploymentName.Namespace)
+			return []reconcile.Request{
+				{
+					NamespacedName: types.NamespacedName{
+						Namespace: agent.Spec.ClusterDeploymentName.Namespace,
+						Name:      agent.Spec.ClusterDeploymentName.Name,
+					},
+				},
+			}
+		} else {
+			return []reconcile.Request{}
```

Comment:
I'm guessing the reason we don't warn about the lack of `agent.Spec.ClusterDeploymentName` in the log is that this is common and to report on it would generate too much noise?

Reply:
We don't warn about it, because this is not really a reason for a warning. Agent without a respective ClusterDeployment is still a valid agent (think about late-binding or some other scenarios).

We could at most `log.Debug` about it, but I don't personally see a big benefit